### PR TITLE
[OASIS] Add Oasis platform

### PR DESCRIPTION
**Pull request by Anonyma**
BASE INTEGRATION `platform/oasis`
   - [x] `base.go`: Init() & Coin() config
   - [x] `model.go`: Platform-specific models	   
   - [x] `client.go`: Platform getter methods (API)
   - [x] `transaction.go`: `NormalizeTx` method to match Atlas model + `GetTxsByAddress` method
   - [x] `block.go`: `CurrentBlockNumber` & `GetBlockByNumber` methods
 
CONFIG
 - [x] `config.yml` + `config/configuration.go`: Add API URL

REST API
- [x] Endpoint to query a list of transactions related to an address
- [x] Endpoint to query a list of transactions from a specific block height
- [x] Endpoint to query latest block height

### CODE REVIEW

File: platform/oasis/base.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+func Init(api string) *Platform {
+	p := &Platform{
+		client: Client{client.InitClient(api, middleware.SentryErrorHandler)},
+	}
+	return p
+}
+
+func (p *Platform) Coin() coin.Coin {
+	return coin.Coins[coin.OASIS]
```
Comment (hewigovens) [inline platform/oasis/base.go:21]:
```suggestion
	return coin.Oasis()
```

File: platform/oasis/transaction.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		Coin:     coin.OASIS,
+		From:     srcTx.From,
+		To:       srcTx.To,
+		Fee:      types.Amount(srcTx.Fee),
+		Date:     srcTx.Date,
+		Block:    srcTx.Block,
+		Status:   status,
+		Error:    srcTx.ErrorMsg,
+		Sequence: srcTx.Sequence,
+		Meta: types.Transfer{
```
Comment (hewigovens) [inline platform/oasis/transaction.go:46]: can you ignore non transfer transactions for now?

File: config.yml [hunk 3]
```diff
@@ -213,6 +213,9 @@ elrond:
 filecoin:
   api: https://api.filscan.io:8700/rpc/v1
   explorer: https://filfox.info
+  
+oasis:
+  api: https://oasis.atlas.zondax.ch
```
Comment (hewigovens) [inline config.yml:218]: can we use this endpoint in production?

File: platform/oasis/transaction.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package oasis
+
+import (
+	"github.com/trustwallet/golibs/coin"
+	"github.com/trustwallet/golibs/types"
+)
+
+func (p *Platform) GetTxsByAddress(address string) (types.TxPage, error) {
```
Comment (hewigovens) [inline platform/oasis/transaction.go:8]: please rebase master, `GetTxsByAddress` should return `types.Txs` now
```suggestion
func (p *Platform) GetTxsByAddress(address string) (types.Txs, error) {
```

File: platform/oasis/transaction.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	txs, err := p.client.GetTrxOfAddress(address)
+	if err != nil {
+		return nil, err
+	}
+
+	return NormalizeTxs(*txs), nil
+}
+
+func NormalizeTxs(srcTxs []Transaction) types.TxPage {
```
Comment (hewigovens) [inline platform/oasis/transaction.go:17]:
```suggestion
func NormalizeTxs(srcTxs []Transaction) types.Txs {
```

**Review by hewigovens — COMMENTED** [review]
do you have sample address to test txs api?

File: platform/oasis/transaction.go (hunk 2 above)
Comment (hewigovens) [reply platform/oasis/transaction.go:46]: @Anonyma can you fix this?

Comment (Anonyma) [reply platform/oasis/transaction.go:46]: @hewigovens yes, only transfer transactions are returned from the API, if that is what you mean

File: config.yml (hunk 3 above)
Comment (Anonyma) [reply config.yml:218]: Please use it for testing for now

> do you have sample address to test txs api?

Here is an example address: `oasis1qpqu7cfadhayekanmrtk8t3v4cx9483gmqu7h6gn`
https://www.oasisscan.com/accounts/detail/oasis1qzt0cma7p9yyuqdz56kkg0t7g9al96glm50aufmy

File: platform/oasis/block.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*types.Block, error) {
+	if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
+		txs := NormalizeTxs(*srcBlock)
+		return &types.Block{
+			Number: num,
+			Txs:    txs,
+		}, nil
+	} else {
```
Comment (vikmeup) [inline platform/oasis/block.go:22]: can you remove else and do `return nil, err`

File: platform/oasis/transaction_test.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+func TestNormalizeTx(t *testing.T) {
+	tests := []struct {
+		name     string
+		filename string
+		wantTx   types.Tx
+	}{
+		{
+			name:     "Test normalize successful transaction without fee",
+			filename: "tx_without_fee.json",
+			wantTx: types.Tx{
```
Comment (vikmeup) [inline platform/oasis/transaction_test.go:20]:
```
Memo: ""
Error: "",
```

you can probably skip it as it's empty by default

**Review by vikmeup — APPROVED** [review]

**Pull request merged**